presto-orc/src/main/java/com/facebook/presto/orc/writer/ColumnWriterFactory.java

```diff
@@ -0,0 +1,88 @@
+package com.facebook.presto.orc.writer;
+
+import com.facebook.presto.common.type.Type;
+import com.facebook.presto.orc.ColumnWriterOptions;
+import com.facebook.presto.orc.DwrfEncryptionInfo;
+import com.facebook.presto.orc.OrcEncoding;
+import com.facebook.presto.orc.metadata.MetadataWriter;
+import com.facebook.presto.orc.metadata.OrcType;
+import org.joda.time.DateTimeZone;
+
+import java.util.List;
+
+import static com.facebook.presto.orc.writer.ColumnWriters.createColumnWriter;
+
+public class ColumnWriterFactory
+{
+    private int columnIndex;
+    private List<OrcType> orcTypes;
+    private Type type;
+    private ColumnWriterOptions columnWriterOptions;
+    private OrcEncoding orcEncoding;
+    private DateTimeZone hiveStorageTimeZone;
+    private DwrfEncryptionInfo dwrfEncryptors;
+    private MetadataWriter metadataWriter;
+
+    ColumnWriterFactory setColumnIndex(int columnIndex)
```

Comment:
This class has many setter methods and if all of them requires to be set, move them to constructor. If the parameters are optional, move them to a builder pattern similar to ColumnWriterOptions.